### PR TITLE
User page changes - 3

### LINKED ISSUE #9: A2a: Add user profile management FE

### PR DESCRIPTION
**Pull request by samyipsh**
- add custom `sign-in` page  (ref repo: https://github.dev/vahid-nejad/custom-login-page-intercepting-routes)
-  unify styles with `sign-up` page
- add some peer prep logos (*circular* and *rectangular* variant)
- difficulty centralizing zod types so just placed it at top of each file for now (can shift to shared library later if need be)

github issues #9, #24

image

### CODE REVIEW

File: prisma/postgres/schema.prisma [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 generator client {
     provider = "prisma-client-js"
-    output = "../../node_modules/@prisma-db-psql/client"
```
Comment (Gabau) [inline prisma/postgres/schema.prisma:6]: Extra space?

File: src/pages/sign-in/index.tsx [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  type GetServerSidePropsContext,
+  InferGetServerSidePropsType,
+} from "next";
+import { PeerPrepRectLogo } from "~/assets/logo";
+import LoginWithCredentials from "~/components/Login";
+import Link from "next/link";
+
+// Signin page with credentials & oauth provider sign in
+type SignInProps = InferGetServerSidePropsType<typeof getServerSideProps>;
+const SignIn = ({ providers }: SignInProps) => {
```
Comment (Gabau) [inline src/pages/sign-in/index.tsx:14]: Why is the background for the pages black with white stripes, are we going to change it to a different color?

Comment (Gabau) [reply src/pages/sign-in/index.tsx:14]: Also have some overflow problems when scrolling
image

**Review by Gabau — COMMENTED** [review]
image
Idt this is intended? Its at the root route

File: prisma/postgres/schema.prisma (hunk 1 above)
Comment (samyipsh) [reply prisma/postgres/schema.prisma:6]: haha that was the default then i removed it then i added it back? idk yea shud remove but doesnt affect behaviour for now

File: src/pages/sign-in/index.tsx (hunk 2 above)
Comment (samyipsh) [reply src/pages/sign-in/index.tsx:14]: :( white stripes? thats not intended >< im still trying to change base css class to black but looking at the sass stuff i held back on that, if base css is black wont have patchy effects on loading

Comment (samyipsh) [reply src/pages/sign-in/index.tsx:14]: thx for mentioning the overscrolling, ill have it the back of my head and try fix it in a diff pr?

File: prisma/postgres/schema.prisma (hunk 1 above)
Comment (Gabau) [reply prisma/postgres/schema.prisma:6]: oh is from formatter ah

File: src/pages/sign-in/index.tsx (hunk 2 above)
Comment (Gabau) [reply src/pages/sign-in/index.tsx:14]: I meant the white stripes caused by border-x in pagelayout. Is that intended?

Comment (Gabau) [reply src/pages/sign-in/index.tsx:14]: It shouldn't be too hard to fix in this pr? I think it is caused by the padding of the sigup component
image
Causing it to grab more space than supposed

> thx for mentioning the overscrolling, ill have it the back of my head and try fix it in a diff pr?

**Review by Nafeij — APPROVED** [review]
Looks sweet!

**Pull request merged**